src/ertai/creature.py

```diff
@@ -0,0 +1,96 @@
+"""
+Class for Creature
+"""
+
+from .card import Card
+from .mana import Mana
+
+
+class Creature(Card):
+    """
+    A class for basic creatures.
+    """
+
+    def __init__(
+        self,
+        title=None,
+        cost=Mana(),
+        tapped=False,
+        power=0,
+        toughness=0,
+    ):
+        """
+        Initiate this creature by giving power, toughness of it.
+
+        Other basic information below of this Creature derived from Card:
+
+        title : title of this creature
+
+        cost : cost to cast this creature
+
+        tapped : tap information of this creature
+
+        Parameters:
+            - power : power  of this creature
+            - toughness : Toughness value of this creature,
+                but this value can be modified in a fight or other condition
+            - base_toughness : Toughness value of this creature,
+                use to store the base toughness of this creature
+        """
+
+        Card.__init__(self, title, cost, tapped)
+
+        self.power = power
+        self.toughness = toughness
+        self.base_toughness = toughness
+        self.state = True
+
+    def __repr__(self):
+        """
+        This is used to print the information about this creature.
+        """
+        return (
+            self.title
+            + "    Cost:"
+            + str(self.cost)
+            + "    Power:"
+            + str(self.power)
+            + "    Toughness:"
+            + str(self.toughness)
+        )
+
+    def check_state(self):
+        """
+        This function is used after a attack happended.
+        If the toughness of this creature drop down to 0 or less,
+        then we set it's state to False which means that this creature is died.
+        """
+        if self.toughness <= 0:
+            self.state = False
```

Comment:
I see what state is for. Let us instead go for `is_alive` and have that as True or False.